### PR TITLE
Special-case an error message to make it clearer

### PR DESCRIPTION
**Pull request by mtrmac**
... and, also, indirectly, confirm the hypothesis in https://github.com/containers/podman/issues/24308 .

### CODE REVIEW

**Comment by mtrmac** [comment]
Confirmed in manual testing:
```console
# bin/podman --storage-driver vfs pull quay.io/libpod/testimage:20241010
Trying to pull quay.io/libpod/testimage:20241010...
Getting image source signatures
Copying blob a3ed95caeb02 done   | 
Error: copying system image from manifest list: reading blob sha256:…: partial layer pull is not supported by the graph driver, and fallback is not allowed: not supported
```

**Review by cgwalters — APPROVED** [review]

File: storage/storage_dest.go [hunk 1]
```diff
@@ -335,6 +335,11 @@ func (s *storageImageDestination) PutBlobPartial(ctx context.Context, chunkAcces
 
 	out, err := s.imageRef.transport.store.PrepareStagedLayer(nil, differ)
 	if err != nil {
+		// Special-case the error message
+		var perr chunked.ErrFallbackToOrdinaryLayerDownload
+		if errors.Is(err, storage.ErrNotSupported) && !errors.As(err, &perr) {
+			err = fmt.Errorf("partial layer pull is not supported by the graph driver, and fallback is not allowed: %w", err)
```
Comment (vrothberg) [inline storage/storage_dest.go:341]: Could we list the graph driver in the error?

Comment (mtrmac) [reply storage/storage_dest.go:341]: Meanwhile I have prepared https://github.com/containers/storage/pull/2140; that is where the error is going to come from in the future.

And c/image does not authoritatively know that the graph driver is the cause.

So I think I’ll simplify this again to just add more context without special-casing.

**Comment by mtrmac** [comment]
Rewritten to just add / improve context. Please review again.

Right now, we should see
```
copying system image from manifest list: partial pull of blob sha256:…: staging a partially-pulled layer: not supported
```

and with https://github.com/containers/storage/pull/2140:
```
copying system image from manifest list: partial pull of blob sha256:…: graph driver vfs does not support partial pull but convert_images requires that
```

**Review by vrothberg — COMMENTED** [review]
Code LGTM

**Comment by rhatdan** [comment]
LGTM

**Pull request merged**